app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/EventsActivity.java

```diff
@@ -27,58 +28,77 @@ public EventsActivity() {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
-        lv = findViewById(R.id.activity_events_list_view);
-        lv.setLayoutManager(new LinearLayoutManager(this));
+        viewPager = (ViewPager) findViewById(R.id.viewPager);
+        setupViewPager(viewPager);
+        Toolbar toolbar = findViewById(R.id.main_toolbar);
+        setSupportActionBar(toolbar);
+        eventTab = (TabLayout) findViewById(R.id.event_tab);
+        eventTab.setupWithViewPager(viewPager);//setting tab over viewpager
+
+        //Implementing tab selected listener over tablayout
+        eventTab.setOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
```

Comment:
setOnTabSelectedListener is marked as deprecated.
We should use addOnTabSelectedListener instead. Simply changing the method name should be sufficient.

https://stackoverflow.com/questions/38498788/setontabselectedlistener-is-deprecated

Reply:
nice suggestion. Thank you